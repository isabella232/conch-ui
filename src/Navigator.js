// src/Navigator.js

import O from "patchinko/constant";

export default update => {
	const pages = {};
	const registerPages = newPages => O(pages, newPages);

	const getCurrentPage = model => {
		// if we don't have auth return the Login page
		if (!model.auth) return getPage({ pageId: "Login" });

		const page = getPage(model);
		if (page) return page;
	};

	const getPage = model => pages[model.pageId];

	const navigateTo = (pageId, params) => {
		const page = getPage({ pageId });

		const navigate = Promise.resolve(O({ pageId }, params));

		if (page && page.navigate) navigate.then(page.navigate);

		navigate.then(update);

		return navigate;
	};

	return {
		getPage,
		getCurrentPage,
		navigateTo,
		registerPages
	};
};
