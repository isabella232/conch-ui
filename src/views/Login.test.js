/* global describe, it, expect */
import m from "mithril";
import stream from "mithril/stream";
import { tidy } from "mithril-jest";

import Navigator from "Navigator";
import LoginView from "views/Login";

describe("Login component", () => {
    const update = stream();
    const navigator = new Navigator(update);
	it("default title should be 'Page'", () => {
        const Login = new LoginView({navigator, update});
		const cmp = m(Login);
		const html = tidy(cmp);
		expect(html).toContain("Login");
		expect(html).toMatchSnapshot();
	});
});
