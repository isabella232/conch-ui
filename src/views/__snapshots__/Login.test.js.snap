// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component default title should be 'Page' 1`] = `
"<section class=\\"hero is-fullheight\\">
  <div class=
  \\"hero-body container has-text-centered column is-4 is-offset-4\\">
    <div class=\\"box\\">
      <h3 class=\\"title\\">
        Login to Conch
      </h3>
      <form>
        <div class=\\"field\\">
          <div class=\\"control\\">
            <input type=\\"email\\" placeholder=\\"Email address\\" class=
            \\"input is-info is-fullwidth is-rounded\\">
          </div>
        </div>
        <div class=\\"field\\">
          <div class=\\"control\\">
            <input type=\\"password\\" placeholder=\\"Password\\" class=
            \\"input is-info is-fullwidth is-rounded\\">
          </div>
        </div><button type=\\"submit\\" class=
        \\"button is-primary is-fullwidth\\">Login</button>
      </form>
    </div>
  </div>
</section>
"
`;
